fire/cli/niv/__init__.py

```diff
@@ -500,16 +500,79 @@ def find_sagsid(sagsgang: pd.DataFrame) -> str:
 def punkter_geojson(
     projektnavn: str,
     punkter: pd.DataFrame,
+    præfix: str = "",
 ) -> None:
     """Skriv punkter/koordinater i geojson-format"""
-    with open(f"{projektnavn}-punkter.geojson", "wt") as punktfil:
+    with open(f"{projektnavn}{præfix}-punkter.geojson", "wt") as punktfil:
         til_json = {
             "type": "FeatureCollection",
             "Features": list(punkt_feature(punkter)),
         }
         json.dump(til_json, punktfil, indent=4)
 
 
+# ------------------------------------------------------------------------------
+def obs_feature(
+    punkter: pd.DataFrame, observationer: pd.DataFrame, antal_målinger: Dict[Tuple, int]
+) -> Dict[str, str]:
+    """Omsæt observationsinformationer til JSON-egnet dict"""
+    for i in range(observationer.shape[0]):
+        fra = observationer.at[i, "Fra"]
+        til = observationer.at[i, "Til"]
+        feature = {
+            "type": "Feature",
+            "properties": {
+                "Fra": fra,
+                "Til": til,
+                "Målinger": antal_målinger[tuple(sorted([fra, til]))],
+                "Afstand": observationer.at[i, "L"],
+                "ΔH": observationer.at[i, "ΔH"],
+                # konvertering, da json.dump ikke uderstøtter int64
+                "Opstillinger": int(observationer.at[i, "Opst"]),
+                "Journal": observationer.at[i, "Journal"],
+                "Type": observationer.at[i, "Type"],
+                "Slukket": observationer.at[i, "Sluk"],
+            },
+            "geometry": {
+                "type": "LineString",
+                "coordinates": [
+                    [punkter.at[fra, "Øst"], punkter.at[fra, "Nord"]],
+                    [punkter.at[til, "Øst"], punkter.at[til, "Nord"]],
+                ],
+            },
+        }
+        yield feature
+
+
+# ------------------------------------------------------------------------------
+def observationer_geojson(
+    projektnavn: str,
+    punkter: pd.DataFrame,
+    observationer: pd.DataFrame,
+    præfix: str = "",
+) -> None:
+    """Skriv observationer til geojson-fil"""
+
+    fra = observationer["Fra"]
+    til = observationer["Til"]
+
+    # Optæl antal frem-og/eller-tilbagemålinger pr. strækning: Vi starter
+    # med en dict med et nul for hver strækning
+    par = [tuple(p) for p in zip(fra, til)]
+    antal_målinger = dict((tuple(sorted(p)), 0) for p in par)
+    # ...og så tæller vi det relevante element op for hver observation
+    for p in par:
+        # Indeksering med tuple(sorted(p)) da set(p) ikke kan hashes
+        antal_målinger[tuple(sorted(p))] += 1
+
+    with open(f"{projektnavn}{præfix}-observationer.geojson", "wt") as obsfil:
```

Comment:
Der er to steder, hvor filnavnet bliver bygget. Jeg ville lave én funktion, der bygger filnavnet, så der kun er ét sted, det skal rettes.

Reply:
Altså her og i `punkter_geojson()`? Det kan jo godt ende ud i en funktion, fx `_geojson_filnavn = lambda projektnavn, infix, variant : f"{projektnavn}{infix}-{variant}.geojson"` og så får vi
```
with open(_geojson_filnavn(projektnavn, præfix, "observationer"), "wt") as obsfil:
# og
with open(_geojson_filnavn(projektnavn, præfix, "punkter"), "wt") as punktfil:
```
 men er der vundet noget særligt ved det? Vi skal stadig specificere om det er punkter eller observationer og om det er en kontrolberegning eller ej.

Reply:
Ja, eller også har vi en linie ovenfor, hvor filnavnet bliver defineret inden brug (jeg véd godt, at det ikke er nødvendigt og giver én linie mere, men det øger læsbarheden).

Ved at definerer filstrukturen ét sted, skal der kun rettes ét sted, hvis laver strukturen om. Skal filen bruges i andre sammenhænge, vil man bare genbruge funktionen i stedet for at skrive nyt og evt. rette fejl, hvis man lavede den forkert.

Det er et simplelt eksempel, men det er stadig en gentagelse, der kan redelegeres til en funktion, så man undgår at hard-code ting mere end én gang:)